polygeist/lib/Dialect/Polygeist/Transforms/DetectReduction.cpp

```diff
@@ -372,13 +425,25 @@ class LoopReductionIter {
         return WalkResult::interrupt();
       }
 
+      ReductionOp Candidate(Load, CandidateStores[0], OtherLoads);
       Operation *MayAliasOp = nullptr;
       if (Loop.getLoopBody()
               .walk([&](Operation *Op) {
                 if (Op == Load || Op == CandidateStores[0] ||
                     llvm::find(OtherLoads, Op) != OtherLoads.end())
                   return WalkResult::advance();
                 if (hasMayAliasEffects(Load, *Op, aliasAnalysis)) {
+                  if (NumVersion <= DetectReductionVersionLimit) {
+                    Operation *LoadOp = Load;
+                    Optional<ValuePair> VP =
+                        polygeist::VersionConditionBuilder::
+                            getAccessorPairUsedByOperationsDefinedOutsideOfLoop(
```

Comment:
I do not think this belongs to the `VersionConditionBuilder` class. Place into a utility sunction instead.

Reply:
I think you should prune Load and Op that do not reference accessor prior to calling the utility function.

Reply:
done.